Core/include/Acts/Propagator/detail/CovarianceEngine.hpp

```diff
@@ -75,14 +96,41 @@ std::tuple<BoundParameters, BoundMatrix, double> boundState(
 ///   - the curvilinear parameters at given position
 ///   - the stepweise jacobian towards it (from last bound)
 ///   - and the path length (from start - for ordering)
-std::tuple<CurvilinearParameters, BoundMatrix, double> curvilinearState(
-    BoundSymMatrix& covarianceMatrix, BoundMatrix& jacobian,
+std::tuple<CurvilinearParameters, Jacobian, double> curvilinearState(
+    Covariance& covarianceMatrix, Jacobian& jacobian,
     FreeMatrix& transportJacobian, FreeVector& derivatives,
-    BoundToFreeMatrix& jacobianLocalToGlobal, const FreeVector& parameters,
+    std::optional<BoundToFreeMatrix>& jacobianLocalToGlobal,
+    ActsMatrixD<8, 7>& jacobianDirToAngle,
+    ActsMatrixD<7, 8>& jacobianAngleToDir, const FreeVector& parameters,
```

Comment:
Could you add comments for the `jacobianDirAngle` and `jacobianAngleToDir`?

Reply:
Fixed